datafusion/src/physical_plan/udf.rs

```diff
@@ -92,6 +95,23 @@ impl ScalarUDF {
             name: name.to_owned(),
             signature: signature.clone(),
             return_type: return_type.clone(),
+            volatility: Default::default(),
```

Comment:
I would move this to the function signature: volatility is an important concept that imo should be part of the `new` (which is where all properties are introduced). Users are free to define a new `new_scalar` with a default volatility if their scalars are systematically volatile.